src/test/java/org/jabref/gui/externalfiles/FileFilterUtilsTest.java

```diff
@@ -0,0 +1,207 @@
+package org.jabref.gui.externalfiles;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+public class FileFilterUtilsTest {
+
+    private final FileFilterUtils fileFilterUtils = new FileFilterUtils();
+    private final LocalDateTime time = LocalDateTime.now();
+
+    @Test
+    public void isDuringLastDayNegativeTest() {
+        assertEquals(fileFilterUtils.isDuringLastDay(time.minusHours(24)), false);
+    }
+
+    @Test
+    public void isDuringLastDayPositiveTest() {
+        assertEquals(fileFilterUtils.isDuringLastDay(time.minusHours(23)), true);
+    }
+
+    @Test
+    public void isDuringLastWeekNegativeTest() {
+        assertEquals(fileFilterUtils.isDuringLastWeek(time.minusDays(7)), false);
+    }
+
+    @Test
+    public void isDuringLastWeekPositiveTest() {
+        assertEquals(fileFilterUtils.isDuringLastWeek(time.minusDays(6).minusHours(23)), true);
+    }
+
+    @Test
+    public void isDuringLastMonthNegativeTest() {
+        assertEquals(fileFilterUtils.isDuringLastMonth(time.minusDays(30)), false);
+    }
+
+    @Test
+    public void isDuringLastMonthPositiveTest() {
+        assertEquals(fileFilterUtils.isDuringLastMonth(time.minusDays(29).minusHours(23)), true);
+    }
+
+    @Test
+    public void isDuringLastYearNegativeTest() {
+        assertEquals(fileFilterUtils.isDuringLastYear(time.minusDays(365)), false);
+    }
+
+    @Test
+    public void isDuringLastYearPositiveTest() {
+        assertEquals(fileFilterUtils.isDuringLastYear(time.minusDays(364).minusHours(23)), true);
+    }
+
+    @Nested
+    class SortingTests {
+
+        private List<Path> files = new ArrayList<Path>();
+        private List<String> expectedSortByDateAscending = new ArrayList<String>();
+        private List<String> expectedSortByDateDescending = new ArrayList<String>();
+        private List<String> wrongOrder = new ArrayList<String>();
+
+        /* Initialize the directory and files used in the sorting tests, and change their last edited dates. */
+        @BeforeEach
+        public void setUp() throws Exception {
+            String dirPath = "src/test/resources/unlinkedFiles";
```

Comment:
Better use a temporary File directory https://www.baeldung.com/junit-5-temporary-directory#1-tempdir-as-a-method-parameter this has the advantage that the dir is automaticaly cleaned